### ISSUE TITLE
[llvm] Build error

### ISSUE THREAD

**Issue by rnett** [microsoft/vcpkg#23077]
**Host Environment**
- OS: Ubuntu 20.04 (WSL 2)
- Compiler: vcpkg version `2022-02-03-457b0309cefb58fae0dcb2e4c06de9bd7582967f`

**To Reproduce**
`./vcpkg install`
with manifest:
```
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
  "name": "my-application",
  "version": "0.0.0",
  "builtin-baseline": "8dbd66f5a7821ced1ed57696b50375a977006813",
  "dependencies": [
    {
      "name": "llvm",
      "default-features": false,
      "features": [
        "mlir"
      ],
      "version>=": "13.0.0"
    }
  ]
}
```

**Failure logs** 
-Cut and paste the appropriate build messages from the console output.
```
A suitable version of cmake was not found (required v3.21.1). Downloading portable cmake v3.21.1...
Downloading cmake...
  https://github.com/Kitware/CMake/releases/download/v3.21.1/cmake-3.21.1-Linux-x86_64.tar.gz -> \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\downloads\cmake-3.21.1-linux-x86_64.tar.gz
Extracting cmake...
Detecting compiler hash for triplet x64-linux...
A suitable version of ninja was not found (required v1.10.2). Downloading portable ninja v1.10.2...
Downloading ninja...
  https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip -> \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\downloads\ninja-linux-1.10.2.zip
Extracting ninja...
The following packages will be built and installed:
    llvm[core,mlir,tools]:x64-linux -> 13.0.0#1 -- \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\buildtrees\versioning_\versions\llvm\b3ebd32f2ff4394d72edeb0f4b804d914a2e3f94
  * vcpkg-cmake[core]:x64-linux -> 2022-01-19 -- \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\buildtrees\versioning_\versions\vcpkg-cmake\b7c050fe60f91dcedef6d87a3f87584151bf8aee
  * vcpkg-cmake-config[core]:x64-linux -> 2022-01-30 -- \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\buildtrees\versioning_\versions\vcpkg-cmake-config\fea8f92ffa5e14c7111fe526f8cc93ecd8f9dbf0
Additional packages (*) will be modified to complete this operation.
Restored 2 packages from /home/rnett/.cache/vcpkg/archives in 1.777 ms. Use --debug to see more details.
Starting package 1/3: vcpkg-cmake-config:x64-linux
Installing package vcpkg-cmake-config[core]:x64-linux...
Elapsed time for package vcpkg-cmake-config:x64-linux: 353 us
Starting package 2/3: vcpkg-cmake:x64-linux
Installing package vcpkg-cmake[core]:x64-linux...
Elapsed time for package vcpkg-cmake:x64-linux: 420.7 us
Starting package 3/3: llvm:x64-linux
Building package llvm[core,mlir,tools]:x64-linux...
-- Installing port from location: \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\buildtrees\versioning_\versions\llvm\b3ebd32f2ff4394d72edeb0f4b804d914a2e3f94
-- Downloading https://github.com/llvm/llvm-project/archive/llvmorg-13.0.0.tar.gz -> llvm-llvm-project-llvmorg-13.0.0.tar.gz...
-- Extracting source \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\downloads\llvm-llvm-project-llvmorg-13.0.0.tar.gz
-- Applying patch 0002-fix-install-paths.patch
-- Applying patch 0003-fix-openmp-debug.patch
-- Applying patch 0004-fix-dr-1734.patch
-- Applying patch 0005-fix-tools-path.patch
-- Applying patch 0007-fix-compiler-rt-install-path.patch
-- Applying patch 0009-fix-tools-install-path.patch
-- Applying patch 0010-fix-libffi.patch
-- Applying patch 0011-fix-libxml2.patch
-- Using source at \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\buildtrees\llvm\src\org-13.0.0-41c002f263.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
CMake Warning at \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg_installed\x64-linux\share\vcpkg-cmake\vcpkg_cmake_configure.cmake:487 (message):
  The following variables are not used in CMakeLists.txt:

      LLDB_ENABLE_CURSES

  Please recheck them and remove the unnecessary options from the
  `vcpkg_cmake_configure` call.

  If these options should still be passed for whatever reason, please use the
  `MAYBE_UNUSED_VARIABLES` argument.
Call Stack (most recent call first):
  buildtrees/versioning_/versions/llvm/b3ebd32f2ff4394d72edeb0f4b804d914a2e3f94/portfile.cmake:236 (vcpkg_cmake_configure)
  scripts/ports.cmake:145 (include)


-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\downloads\tools\cmake-3.21.1-linux\cmake-3.21.1-linux-x86_64\bin\cmake --build . --config Debug --target install -- -v -j17
    Working Directory: \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\buildtrees\llvm\x64-linux-dbg
    See logs for more information:
      \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg\buildtrees\llvm\install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg_installed\x64-linux\share\vcpkg-cmake\vcpkg_cmake_build.cmake:111 (vcpkg_execute_build_process)
  \\wsl$\Ubuntu-20.04\home\rnett\Projects\Copies\kotlin-mlir-emitter\build\vcpkg_installed\x64-linux\share\vcpkg-cmake\vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  buildtrees/versioning_/versions/llvm/b3ebd32f2ff4394d72edeb0f4b804d914a2e3f94/portfile.cmake:261 (vcpkg_cmake_install)
  scripts/ports.cmake:145 (include)


Error: Building package llvm:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+llvm
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[llvm]+Build+error
including:
  package: llvm[core,mlir,tools]:x64-linux -> 13.0.0#1
    vcpkg-tool version: 2022-02-03-457b0309cefb58fae0dcb2e4c06de9bd7582967f
    vcpkg-scripts version: 8dbd66f5a 2022-02-11 (2 days ago)

Additionally, attach any relevant sections from the log files above.
```
-Please attach any additional failure logs mentioned in the console output.
[install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/8056937/install-x64-linux-dbg-out.log)

**Comment by rnett** [comment]
Seems to be related to the LLVM 13, I was able to fix this by downgrading to 12.0.1 (builtin baseline `e6ed80cfca60da5ad016b28764818be7b7b0b77f`).

**Comment by PhoebeHui** [comment]
Failures:
```
FAILED: bin/mlir-linalg-ods-yaml-gen 
: && /usr/bin/c++ -fPIC -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -g -Wl,-rpath-link,/home/rnett/Projects/Copies/kotlin-mlir-emitter/build/vcpkg/buildtrees/llvm/x64-linux-dbg/./lib tools/mlir/tools/mlir-linalg-ods-gen/CMakeFiles/mlir-linalg-ods-yaml-gen.dir/mlir-linalg-ods-yaml-gen.cpp.o -o bin/mlir-linalg-ods-yaml-gen  -Wl,-rpath,"\$ORIGIN/../lib"  lib/libLLVMCored.a  lib/libLLVMSupportd.a  -lpthread  lib/libMLIRIRd.a  lib/libMLIRSupportd.a  lib/libMLIRParserd.a  lib/libLLVMBinaryFormatd.a  lib/libLLVMRemarksd.a  lib/libLLVMBitstreamReaderd.a  lib/libMLIRIRd.a  lib/libMLIRSupportd.a  lib/libLLVMSupportd.a  -lrt  -ldl  -lm  lib/libLLVMDemangled.a  -lpthread && :
/usr/bin/ld: lib/libMLIRIRd.a(Verifier.cpp.o): in function `mlir::LogicalResult mlir::failableParallelForEach<mlir::Operation**, (anonymous namespace)::OperationVerifier::verifyOpAndDominance(mlir::Operation&)::{lambda(mlir::Operation*)#1}>(mlir::MLIRContext*, mlir::Operation**, mlir::MLIRContext*, (anonymous namespace)::OperationVerifier::verifyOpAndDominance(mlir::Operation&)::{lambda(mlir::Operation*)#1}&&)':
/home/rnett/Projects/Copies/kotlin-mlir-emitter/build/vcpkg/buildtrees/llvm/src/org-13.0.0-41c002f263.clean/mlir/include/mlir/IR/Threading.h:47: undefined reference to `llvm::ThreadPool::isWorkerThread() const'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```
